ui/src/components/NodePageMetricsTab.js

```diff
@@ -87,98 +85,98 @@ const NodePageMetricsTab = (props) => {
       sampleFrequency,
     );
 
-  const cpuUsageOperated = operateMetricRawData(nodeStats?.cpuUsage[0]?.values);
-  const systemLoadOperated = operateMetricRawData(
-    nodeStats?.systemLoad[0]?.values,
-  );
-  const memoryOperated = operateMetricRawData(nodeStats?.memory[0]?.values);
-  const iopsWriteOperated = operateMetricRawData(
-    nodeStats?.iopsWrite[0]?.values,
-  );
-  const iopsReadOperated = operateMetricRawData(nodeStats?.iopsRead[0]?.values);
-  const controlPlaneNetworkBandwidthInOperated = operateMetricRawData(
-    nodeStats?.controlPlaneNetworkBandwidthIn[0]?.values,
-  );
-  const controlPlaneNetworkBandwidthOutOperated = operateMetricRawData(
-    nodeStats?.controlPlaneNetworkBandwidthOut[0]?.values,
-  );
-  const workloadPlaneNetworkBandwidthInOperated = operateMetricRawData(
-    nodeStats?.workloadPlaneNetworkBandwidthIn[0]?.values,
-  );
-  const workloadPlaneNetworkBandwidthOutOperated = operateMetricRawData(
-    nodeStats?.workloadPlaneNetworkBandwidthOut[0]?.values,
-  );
-
-  const cpuUsageData = cpuUsageOperated?.map((slot) => {
-    return {
-      date: fromUnixTimestampToDate(slot[0]),
-      y: slot[1],
-    };
-  });
-  const systemLoadData = systemLoadOperated?.map((slot) => {
+  const nodeStatsOperated = Object.keys(nodeStats).map((metricName) => {
     return {
-      date: fromUnixTimestampToDate(slot[0]),
-      y: slot[1],
-    };
-  });
-  const memoryData = memoryOperated?.map((slot) => {
-    return {
-      date: fromUnixTimestampToDate(slot[0]),
-      y: slot[1],
+      metrics: metricName,
+      data: operateMetricRawData(nodeStats[metricName][0]?.values),
     };
   });
 
-  const nodeIOPSWriteData = iopsWriteOperated?.map((slot) => {
-    return {
-      date: fromUnixTimestampToDate(slot[0]),
-      write: slot[1],
-      type: 'write',
-    };
-  });
-  const nodeIOPSReadData = iopsReadOperated?.map((slot) => {
-    return {
-      date: fromUnixTimestampToDate(slot[0]),
-      read: slot[1],
-      type: 'read',
-    };
-  });
-  const nodeControlPlaneNetworkBandwidthInData = controlPlaneNetworkBandwidthInOperated?.map(
-    (slot) => {
+  const cpuUsageData = nodeStatsOperated
+    ?.find((obj) => obj.metrics === 'cpuUsage')
+    ?.data?.map((slot) => {
+      return {
+        date: fromUnixTimestampToDate(slot[0]),
+        y: slot[1],
+      };
+    });
```

Comment:
Weren't trying to reduce duplication? :stuck_out_tongue_winking_eye:

Instead of doing this mapping after building `nodeStatsOperated`, you could do it during construction:

```javascript
  const nodeStatsOperated = Object.keys(nodeStats).reduce((acc, metricName) => {
    const data = operateMetricRawData(nodeStats[metricName][0]?.values);
    acc[metricName] = data.map((slot) => ({
      date: fromUnixTimestampToDate(slot[0]),
      y: slot[1],
    }));
    return acc;
  }, {});
```

And then consume it like:

```javascript
<LineChart
  id={'node_cpu_usage_id'}
  data={nodeStatsOperated['cpuUsage']}
  {...rest}
/>
```

Reply:
Hello @gdemonet 
Was trying to reduce the duplication here too. But we have different return keys when it comes to the graphs with multiple line (in/out, read/write)
Please see:
```
  const nodeWorkloadPlaneNetworkBandwidthOutData = nodeStatsOperated
    ?.find((obj) => obj.metrics === 'workloadPlaneNetworkBandwidthOut')
    ?.data?.map((slot) => {
      return {
        date: fromUnixTimestampToDate(slot[0]),
        out: slot[1],
        type: 'out',
      };
    });
```

Reply:
Oh right, missed that one. There still are some ways to reduce this:

- each `.find((obj) => obj.metrics === someKey)` could be simplified if `nodeStatsOperated` was a map instead of an array:

```javascript
const nodeWorkloadPlaneNetworkBandwidthOutData = nodeStatsOperated
  ?.['workloadPlaneNetworkBandwidthOut']?.map((slot) => { ... })
```

- if you want to push it further, adding a `type` key can be done easily by inferring it from the `metricName`:

```javascript
// First variant, assumes a given format for `metricName`
const nodeStatsOperated = Object.keys(nodeStats).reduce((acc, metricName) => {
  const data = operateMetricRawData(nodeStats[metricName][0]?.values);

  let extra = {};
  if (metricName.endsWith('Out')) extra.type = 'out';
  else if (metricName.endsWith('In')) extra.type = 'in';
  else if (metricName.endsWith('Read')) extra.type = 'read';
  else if (metricName.endsWith('Write')) extra.type = 'write';

  acc[metricName] = data.map((slot) => ({
    date: fromUnixTimestampToDate(slot[0]),
    y: slot[1],
    ...extra
  }));
  return acc;
}, {});

// Second variant, stores the mapping between types and metric names
const typedMetrics = {
  iopsRead: 'read',
  iopsWrite: 'write',
  controlPlaneNetworkBandwidthIn: 'in',
  controlPlaneNetworkBandwidthOut: 'out',
  workloadPlaneNetworkBandwidthIn: 'in',
  workloadPlaneNetworkBandwidthOut: 'out',
};

const nodeStatsOperated = Object.keys(nodeStats).reduce((acc, metricName) => {
  const data = operateMetricRawData(nodeStats[metricName][0]?.values);

  let extra = {};
  const metricType = typedMetrics[metricName];
  if (metricType !== undefined) extra.type = metricType;

  acc[metricName] = data.map((slot) => ({
    date: fromUnixTimestampToDate(slot[0]),
    y: slot[1],
    ...extraType
  }));
  return acc;
}, {});
```